File: 7kyu/Reverser/Python/Reverser.py
def reverse(n):
    reversed_number = 0
    negative = n < 0
    n = abs(n)

    while n != 0:
        reversed_number = reversed_number * 10 + n % 10
        n //= 10

    return -reversed_number if negative else reversed_number

# Example usage:
print(reverse(123))  # Output: 321
print(reverse(-456)) # Output: -654
print(reverse(1000)) # Output: 1

File: 7kyu/Reverser/JavaScript/Reverser.js
function reverse(n) {
    let reversedNumber = 0;
    let negative = n < 0;
    n = Math.abs(n);

    while (n > 0) {
        let lastDigit = n % 10;
        reversedNumber = reversedNumber * 10 + lastDigit;
        n = Math.floor(n / 10);
    }

    return negative ? -reversedNumber : reversedNumber;
}

// Example usage:
console.log(reverse(123));  // Output: 321
console.log(reverse(-456)); // Output: -654
console.log(reverse(1000)); // Output: 1
